File: solutions/cpp/grade-school/1/grade_school.cpp
#include "grade_school.h"

#include <map>
#include <algorithm>
#include <vector>
#include <string>

using namespace std;

namespace grade_school {

    void school::add(string student, int grade){
        roster_[grade].push_back(student);   
        std::sort(roster_[grade].begin(), roster_[grade].end());
    }

    const vector<string> school::grade(int g) const {
        return roster_.find(g)->second;
    }

    const map<int, vector<string>> &school::roster() const {
        return roster_;
    }

    
    
}  // namespace grade_school

File: solutions/dart/grade-school/2/lib/grade_school.dart
class GradeSchool {
  Map<String, int> myRoster =  {};

  List<bool> add(List<(String, int)> students){
    var result = <bool>[];
    students.forEach((kv){
      if(!myRoster.containsKey(kv.$1)){
        myRoster[kv.$1] = kv.$2; 
        result.add(true);
      }
      else result.add(false);      
    });
    return result;
  }

  List<String> grade(int g) =>  
   myRoster.entries
      .where((e)=>e.value == g)
      .map((e)=>e.key).toList()..sort();

   List<String> roster() => 
    (myRoster.values.toSet().toList()..sort())
     .map((n) => grade(n))
     .fold(<String>[], (v,e) => v + e);
 
}
